<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Preparing `MDSvis` input objects from a distance matrix and sample properties • MDSvis</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Preparing `MDSvis` input objects from a distance matrix and sample properties">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MDSvis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/MDSvis.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/MDSvis_Input_From_Distance.html">Preparing `MDSvis` input objects from a distance matrix and sample properties</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/UCLouvain-CBIO/MDSvis/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Preparing `MDSvis` input objects from a distance matrix and sample properties</h1>
                        <h4 data-toc-skip class="author">Philippe
Hauchamps</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/UCLouvain-CBIO/MDSvis/blob/main/vignettes/MDSvis_Input_From_Distance.Rmd" class="external-link"><code>vignettes/MDSvis_Input_From_Distance.Rmd</code></a></small>
      <div class="d-none name"><code>MDSvis_Input_From_Distance.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette shows an example of building input objects for
      using <code>MDSvis</code>. Here the starting point is: (1) a
      pairwise distance matrix, where each row and each column
      correspond to one sample, and (2) additional sample specific
      properties. This vignette is distributed under a CC BY-SA 4.0
      license.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This short vignette is complementary to the <a href="MDSvis.html">main</a> <code>MDSvis</code> package vignette, which
provides an overview of <code>MDSvis</code> Shiny application (app)
features.</p>
<p>Here we focus on ‘how to’ create some input objects, which need to be
provided to the Shiny app. In this vignette, we show that we can take
advantage of the <em>Multi Dimensional Scaling (MDS)</em> interactive
visualisations, starting from any distance matrix, and some sample
specific properties.</p>
</div>
<div class="section level2">
<h2 id="installation-and-loading-dependencies">Installation and loading dependencies<a class="anchor" aria-label="anchor" href="#installation-and-loading-dependencies"></a>
</h2>
<p>To install this package, start R and enter (un-commented):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("devtools", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("devtools")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># devtools::install_github(repo = 'https://github.com/UCLouvain-CBIO/MDSvis')</span></span></code></pre></div>
<p>We now load the packages needed in the current vignette:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://uclouvain-cbio.github.io/CytoMDS" class="external-link">CytoMDS</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://uclouvain-cbio.github.io/MDSvis">MDSvis</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dataset">Dataset<a class="anchor" aria-label="anchor" href="#dataset"></a>
</h2>
<p>For this vignette, we use the <em>frog skin active peptide
(fsap)</em> dataset. It consists of 226 samples, each sample
representing here a amino acid sequence corresponding to a known anuran
cytolytic peptide.</p>
<p>Distance between these peptides can be obtained, by aligning them and
computing an appropriate dissimilarity metric between the corresponding
amino acid sequences. For details, see the <a href="#fsap_methods">fsap
dataset methods section</a> below.</p>
<p>On top of the peptide distances, peptide properties have also been
collected:<br>
- Uniprot accession number<br>
- family, genius, species<br>
- some physicochemical parameters</p>
<p>In the following code, we import the two text files of the
dataset:<br>
- the distance matrix file (imported as a R <code>matrix</code>)<br>
- the sample properties file (imported as a R
<code>data.frame</code>)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># inputs</span></span>
<span><span class="va">inputDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"MDSvis"</span><span class="op">)</span></span>
<span><span class="va">distPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"fsap_mature_226_distances.txt"</span><span class="op">)</span></span>
<span><span class="va">propPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"fsap_mature_226_properties.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># reading distance matrix from disk</span></span>
<span><span class="va">distData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">distPath</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">distData</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">distData</span><span class="op">$</span><span class="va">X</span></span>
<span><span class="va">distData</span><span class="op">$</span><span class="va">X</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">distMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">distData</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># reading properties</span></span>
<span><span class="va">propDF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">propPath</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generation-of-input-files-for-the-shiny-application">Generation of input files for the Shiny application<a class="anchor" aria-label="anchor" href="#generation-of-input-files-for-the-shiny-application"></a>
</h2>
<p>We still need to pre-process the data properties and split them into
two parts: - the variables that will be used to annotate the plots, or
associate colour, shape, faceting etc. These variable are mostly factor
variables. We extract them into the <em>phenoData</em> object - which
needs to be a <code>data.frame</code>. - the continuous variables that
can be exploited as sample <em>statistics</em> in the bi-plots. These
need to be included in a list of statistics object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># split properties between factors and continuous variables (stats)</span></span>
<span></span>
<span><span class="va">statMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">statList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>quant_properties <span class="op">=</span> <span class="va">statMat</span><span class="op">)</span></span>
<span></span>
<span><span class="va">propDF</span> <span class="op">&lt;-</span> <span class="va">propDF</span><span class="op">[</span>, <span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">propDF</span><span class="op">$</span><span class="va">family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">Sequence_Name</span>, start <span class="op">=</span> <span class="fl">1</span>, stop <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">propDF</span><span class="op">$</span><span class="va">genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">Sequence_Name</span>, start <span class="op">=</span> <span class="fl">1</span>, stop <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">propDF</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">Sequence_Name</span>, start <span class="op">=</span> <span class="fl">1</span>, stop <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Below are frequency tables for peptides belonging to the different
families, genius and species.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">family</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## HYL MIC RAN </span></span>
<span><span class="co">##  43   7 176</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">genus</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## HYL_AG HYL_CR HYL_HP HYL_HY HYL_LD HYL_LI HYL_PH HYL_PL HYL_SP MIC_KA RAN_AM </span></span>
<span><span class="co">##      9      2      2      1      2      7     17      2      1      7     10 </span></span>
<span><span class="co">## RAN_BA RAN_FE RAN_GL RAN_HD RAN_LB RAN_NA RAN_OD RAN_PE RAN_RA RAN_SY RAN_TH </span></span>
<span><span class="co">##     15      1      2      1     15      3     63     19     16     30      1</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">propDF</span><span class="op">$</span><span class="va">species</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## HYL_AGca HYL_AGda HYL_CRca HYL_HPci HYL_HPra HYL_HYsi HYL_LDpu HYL_LIfa </span></span>
<span><span class="co">##        6        3        2        1        1        1        2        4 </span></span>
<span><span class="co">## HYL_LIin HYL_PHbi HYL_PHhy HYL_PLth HYL_SPla MIC_KAma MIC_KAse RAN_AMha </span></span>
<span><span class="co">##        3       10        7        2        1        2        5        4 </span></span>
<span><span class="co">## RAN_AMlo RAN_BAad RAN_BApl RAN_FEca RAN_GLru RAN_HDma RAN_LBca RAN_LBpi </span></span>
<span><span class="co">##        6        5       10        1        2        1        7        8 </span></span>
<span><span class="co">## RAN_NApa RAN_NApl RAN_ODan RAN_ODli RAN_PEle RAN_PEni RAN_RAbo RAN_RAch </span></span>
<span><span class="co">##        2        1       47       16        4       15        5       11 </span></span>
<span><span class="co">## RAN_SYni RAN_SYsp RAN_THkw </span></span>
<span><span class="co">##       13       17        1</span></span></code></pre>
<p>From the distance matrix, we now need to calculate the MDS projection
object, which will be used as main input in the Shiny app.<br>
This is done using the <code>CytoMDS</code> package. Here we set the
number of projection dimensions to 10. See the documentation of
<code><a href="https://uclouvain-cbio.github.io/CytoMDS/reference/computeMetricMDS.html" class="external-link">CytoMDS::computeMetricMDS()</a></code> for details on other possible
options.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MDSObj</span> <span class="op">&lt;-</span> <span class="fu">CytoMDS</span><span class="fu">::</span><span class="fu"><a href="https://uclouvain-cbio.github.io/CytoMDS/reference/computeMetricMDS.html" class="external-link">computeMetricMDS</a></span><span class="op">(</span>pwDist <span class="op">=</span> <span class="va">distMat</span>, </span>
<span>                                    nDim <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">MDSObj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MDS object containing MDS projection (using Smacof algorithm)  data:</span></span>
<span><span class="co">## Nb of dimensions:  10 </span></span>
<span><span class="co">## Nb of points:  226 </span></span>
<span><span class="co">## Stress:  0.156626 </span></span>
<span><span class="co">## Pseudo RSquare:  0.829141 </span></span>
<span><span class="co">## Goodness of fit:  0.975468</span></span></code></pre>
<p>Finally, we can save the newly created objects as .rds files on
disk.<br>
(Uncomment the corresponding code to do it).<br>
These files can now be selected within the shiny app for
visualization.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># outputs (for shiny app)</span></span>
<span><span class="va">MDSRDSPath</span> <span class="op">&lt;-</span> <span class="st">"fsap_mature_MDS.rds"</span></span>
<span><span class="va">statRDSPath</span> <span class="op">&lt;-</span> <span class="st">"fsap_mature_stats.rds"</span></span>
<span><span class="va">propRDSPath</span> <span class="op">&lt;-</span> <span class="st">"fsap_mature_props.rds"</span></span>
<span></span>
<span><span class="co"># saving objects </span></span>
<span><span class="co">#saveRDS(MDSObj, file = MDSRDSPath)</span></span>
<span><span class="co">#saveRDS(statList, file = statRDSPath)</span></span>
<span><span class="co">#saveRDS(propDF, file = propRDSPath)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization-of-the-mds-projection">Visualization of the MDS projection<a class="anchor" aria-label="anchor" href="#visualization-of-the-mds-projection"></a>
</h2>
<p>The <code>MDSvis</code> function <code>run_app</code> launches the
interactive Shiny app. From here, you can take a look at the <a href="MDSvis.html">main</a> <code>MDSvis</code> vignette for an overview
of the Shiny app features.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#MDSvis::run_app()</span></span></code></pre></div>
<p>Here below is a screen shot from the Shiny app, displaying a view of
the MDS projection of the <code>fsap</code> dataset.</p>
<p><img src="images/fsap_view.png"></p>
</div>
<div class="section level2">
<h2 id="fsap-dataset-methods">fsap dataset methods<a class="anchor" aria-label="anchor" href="#fsap-dataset-methods"></a>
</h2>
<p><a name="fsap_methods"> </a></p>
<div class="section level3">
<h3 id="data-mining-and-assembly-of-a-reference-dataset">Data mining and assembly of a reference dataset<a class="anchor" aria-label="anchor" href="#data-mining-and-assembly-of-a-reference-dataset"></a>
</h3>
<p>For the construction of an initial dataset of known anuran cytolytic
peptides, sequences belonging to the “frog skin active peptide”
superfamily (FSAP; InterPro: IPR016322) were collected from the Uniprot
database and relevant literature. This superfamily has been shown to
have evolved at the latest in an early ancestor of the Neobatrachia
(Vanhoye et al. 2003), a clade including almost 95% of anuran species
(Frost 2024). A total of 226 precursor protein sequences representing
FSAP was selected in order to encompass maximal phylogenetic diversity
and sequence diversity. (Note: the superfamily Ranoidea contains the
families Ranidae and Microhylidae. These FSAP have been differentiated
using the prefixes RAN_and MIC_. All FSAP from Hyloidea include the
prefix HYL_. Each prefix is followed by a letter code to identify the
genus and species from which the FSAP was identified, as well as its
Uniprot accession.)</p>
</div>
<div class="section level3">
<h3 id="phylogenetic-reconstruction-of-the-fsap-superfamily">Phylogenetic reconstruction of the FSAP superfamily<a class="anchor" aria-label="anchor" href="#phylogenetic-reconstruction-of-the-fsap-superfamily"></a>
</h3>
<p>Multiple sequence alignment of a dataset containing 225 FSAP
precursor protein sequences (52 from Hyloidea and 183 from Ranoidea) was
carried out using the E-INS-I algorithm in MAFFT v.7 (Katoh and Standley
2013) and manually curated to remove large gaps and poorly aligned
regions in Mesquite v.3.81 (Maddison 2023). Reconstruction of FSAP
phylogeny involved the use of two different methods: Bayesian
phylogenetic inference and maximum likelihood (ML) estimation, as
implemented in MrBayes v.3.2.7. (Ronquist et al. 2012) and RAxML
(Stamatakis 2014), respectively. Results of the two approaches were
compared to verify the absence of major conflicts related to the choice
of method and to obtain a single consensus evolutionary synthesis.</p>
</div>
<div class="section level3">
<h3 id="identification-of-mature-fsap-sequences">Identification of mature FSAP sequences<a class="anchor" aria-label="anchor" href="#identification-of-mature-fsap-sequences"></a>
</h3>
<p>Comparison of unannotated or newly predicted precursor proteins with
previously characterised FSAP allowed us to predict the position and
predicted cleavage sites of mature peptide sequences. The most common
cleavage recognition motifs for FSAP are as follows: -RXXR-, -KR-, -RR-.
A total of 226 mature FSAP were thus identified and used for subsequent
analyses (the discrepancy in total number of sequences arises from the
fact that multiple cytolytic peptides may be cleaved from the same
precursor protein; in this dataset, the relevant sequences are
differentiated by a numbered suffix, and pertain to HYL_LIfa_B5LUQ3_1
and HYL_LIfa_B5LUQ3_2).</p>
</div>
<div class="section level3">
<h3 id="pairwise-alignments-using-primary-structure-and-hydrophobicity">Pairwise alignments using primary structure and hydrophobicity<a class="anchor" aria-label="anchor" href="#pairwise-alignments-using-primary-structure-and-hydrophobicity"></a>
</h3>
<p>Pairwise alignments of mature FSAP sequences were generated with
AlignMe 1.2 (Khafizov et al. 2010; Stamm et al. 2013) using a combined
scale that incorporates the Hessa, White and von Heinje hydropathy scale
(HWvH; weight: 2.6) (Hessa et al. 2007) and the VTML substitution matrix
(weight: 0.5), according to the developer’s recommendations. Gap
penalties were similarly and were set as follows: below threshold gap
opening penalty: 18.0, below threshold gap extension penalty: 1.3, above
threshold gap opening penalty: 11.0, above threshold gap extension
penalty 4.4: thresholds for penalties: 0; termini gap opening penalty
13.1; termini gap extension penalty: 0.9 (Stamm et al. 2014). Given the
short lengths of mature FSAP sequences, window size was set to 0 to
prevent a smoothing effect. Absolute differences between the
hydrophobicity measurements generated for each peptide pair were
calculated and used to construct a distance, or dissimilarity
matrix.</p>
</div>
<div class="section level3">
<h3 id="physicochemical-parameters-distinguishing-fsap">Physicochemical parameters distinguishing FSAP<a class="anchor" aria-label="anchor" href="#physicochemical-parameters-distinguishing-fsap"></a>
</h3>
<p>The following parameters were calculated for each mature FSAP
sequence: 1) peptide length (number of amino acids) 2) grand average of
hydropathy, or GRAVY index (obtained by dividing the sum of hydropathy
values of all amino acids by total sequence length), to determine the
hydrophobicity/hydrophilicity of the protein; a positive index indicates
a more hydrophobic character, whereas a negative index indicates a more
hydrophilic character 3) hydrophobic moment (measure of the
amphipathicity of a structure, or asymmetry of hydrophobicity, by
quantifying the distribution of hydrophobic and hydrophilic residues
along its length; particularly relevant for α-helices) 4) isoelectric
point, or pI (pH at which the peptide carries a net electrical charge of
zero) 5) molar mass 6) net charge at pH 7 7) cysteine content (%) 8)
ratio of lysines to arginines (e.g., changes in this ratio in α-helical
peptides can affect the hydrophobicity required for membrane
perturbation)</p>
</div>
</div>
<div class="section level2">
<h2 id="acknowledgement">Acknowledgement<a class="anchor" aria-label="anchor" href="#acknowledgement"></a>
</h2>
<p>We would like to thank Shabnam Zaman, from the Amphibian Evolution
Lab of the Vrije Universiteit Brussel (VUB), for providing us with the
<code>fsap</code> dataset, as well as the associated Methods
section.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] MDSvis_0.1.6        CytoMDS_1.5.0       Biobase_2.69.0     </span></span>
<span><span class="co">## [4] BiocGenerics_0.55.0 generics_0.1.4      BiocStyle_2.37.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] Rdpack_2.6.4         polynom_1.4-1        gridExtra_2.3       </span></span>
<span><span class="co">##   [4] rlang_1.1.6          magrittr_2.0.3       e1071_1.7-16        </span></span>
<span><span class="co">##   [7] matrixStats_1.5.0    compiler_4.5.0       gdata_3.0.1         </span></span>
<span><span class="co">##  [10] systemfonts_1.2.3    vctrs_0.6.5          stringr_1.5.1       </span></span>
<span><span class="co">##  [13] pkgconfig_2.0.3      shape_1.4.6.1        fastmap_1.2.0       </span></span>
<span><span class="co">##  [16] backports_1.5.0      promises_1.3.2       ncdfFlow_2.55.0     </span></span>
<span><span class="co">##  [19] rmarkdown_2.29       nloptr_2.2.1         graph_1.87.0        </span></span>
<span><span class="co">##  [22] ragg_1.4.0           purrr_1.0.4          xfun_0.52           </span></span>
<span><span class="co">##  [25] glmnet_4.1-8         jomo_2.7-6           cachem_1.1.0        </span></span>
<span><span class="co">##  [28] jsonlite_2.0.0       flowWorkspace_4.21.0 later_1.4.2         </span></span>
<span><span class="co">##  [31] pan_1.9              parallel_4.5.0       broom_1.0.8         </span></span>
<span><span class="co">##  [34] cluster_2.1.8.1      R6_2.6.1             bslib_0.9.0         </span></span>
<span><span class="co">##  [37] stringi_1.8.7        RColorBrewer_1.1-3   smacof_2.1-7        </span></span>
<span><span class="co">##  [40] boot_1.3-31          rpart_4.1.24         jquerylib_0.1.4     </span></span>
<span><span class="co">##  [43] Rcpp_1.0.14          bookdown_0.43        iterators_1.0.14    </span></span>
<span><span class="co">##  [46] knitr_1.50           base64enc_0.1-3      weights_1.0.4       </span></span>
<span><span class="co">##  [49] flowCore_2.21.0      httpuv_1.6.16        Matrix_1.7-3        </span></span>
<span><span class="co">##  [52] nnls_1.6             splines_4.5.0        nnet_7.3-20         </span></span>
<span><span class="co">##  [55] tidyselect_1.2.1     rstudioapi_0.17.1    yaml_2.3.10         </span></span>
<span><span class="co">##  [58] doParallel_1.0.17    codetools_0.2-20     lattice_0.22-7      </span></span>
<span><span class="co">##  [61] tibble_3.2.1         plyr_1.8.9           shiny_1.10.0        </span></span>
<span><span class="co">##  [64] evaluate_1.0.3       foreign_0.8-90       desc_1.4.3          </span></span>
<span><span class="co">##  [67] survival_3.8-3       proxy_0.4-27         pillar_1.10.2       </span></span>
<span><span class="co">##  [70] BiocManager_1.30.25  mice_3.17.0          checkmate_2.3.2     </span></span>
<span><span class="co">##  [73] foreach_1.5.2        stats4_4.5.0         ellipse_0.5.0       </span></span>
<span><span class="co">##  [76] reformulas_0.4.1     shinyjs_2.1.0        plotly_4.10.4       </span></span>
<span><span class="co">##  [79] S4Vectors_0.47.0     ggplot2_3.5.2        ggcyto_1.37.0       </span></span>
<span><span class="co">##  [82] scales_1.4.0         minqa_1.2.8          gtools_3.9.5        </span></span>
<span><span class="co">##  [85] xtable_1.8-4         class_7.3-23         glue_1.8.0          </span></span>
<span><span class="co">##  [88] Hmisc_5.2-3          lazyeval_0.2.2       tools_4.5.0         </span></span>
<span><span class="co">##  [91] hexbin_1.28.5        data.table_1.17.2    lme4_1.1-37         </span></span>
<span><span class="co">##  [94] fs_1.6.6             XML_3.99-0.18        grid_4.5.0          </span></span>
<span><span class="co">##  [97] plotrix_3.8-4        tidyr_1.3.1          RProtoBufLib_2.21.0 </span></span>
<span><span class="co">## [100] rbibutils_2.3        CytoPipeline_1.9.0   colorspace_2.1-1    </span></span>
<span><span class="co">## [103] nlme_3.1-168         htmlTable_2.4.3      Formula_1.2-5       </span></span>
<span><span class="co">## [106] cli_3.6.5            textshaping_1.0.1    cytolib_2.21.0      </span></span>
<span><span class="co">## [109] viridisLite_0.4.2    dplyr_1.1.4          Rgraphviz_2.53.0    </span></span>
<span><span class="co">## [112] gtable_0.3.6         sass_0.4.10          digest_0.6.37       </span></span>
<span><span class="co">## [115] wordcloud_2.6        htmlwidgets_1.6.4    farver_2.1.2        </span></span>
<span><span class="co">## [118] htmltools_0.5.8.1    pkgdown_2.1.2.9000   lifecycle_1.0.4     </span></span>
<span><span class="co">## [121] httr_1.4.7           mitml_0.4-5          mime_0.13           </span></span>
<span><span class="co">## [124] MASS_7.3-65</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrea Vicini, Philippe Hauchamps, Laurent Gatto.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
